app/jobs/scheduled/delete_hidden_queries.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Jobs
+  class DeleteHiddenQueries < ::Jobs::Scheduled
+    every 7.days
+
+    def execute(args)
+      return unless SiteSetting.data_explorer_enabled
+
+      DataExplorer::Query.where(hidden: true).each do |query|
```

Comment:
Some sites have stacked up 10,000+ hidden queries, so we should probably do this in batches... Or alternatively do the `last_run_at` and `updated_at` checks in SQL, and use `.delete_all`

Reply:
Good idea. Will make it so.